src/test/java/com/google/edith/ReceiptFileHandlerServiceTest.java

```diff
@@ -0,0 +1,130 @@
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class ReceiptFileHandlerServiceTest {
+
+  private final LocalServiceTestHelper testHelper =
+      new LocalServiceTestHelper(
+          new LocalBlobstoreServiceTestConfig(), new LocalDatastoreServiceTestConfig());
+
+  private ReceiptFileHandlerService receiptFileHandlerService;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    testHelper.setUp();
+    receiptFileHandlerService = new ReceiptFileHandlerService(blobstoreService);
+  }
+
+  @After
+  public void tearDown() {
+    testHelper.tearDown();
+  }
+
+  @Mock BlobstoreService blobstoreService;
+
+  @Mock HttpServletRequest request;
+
+  @Mock HttpServletResponse response;
+
+  @Test
+  public void checks_ifBlobDidNotUpload_returnsNull() throws IOException {
+    Map<String, List<FileInfo>> emptyMap = Collections.emptyMap();
+
+    when(blobstoreService.getFileInfos(request)).thenReturn(emptyMap);
+    Optional<List<FileInfo>> uploadedFileInfo =
+        receiptFileHandlerService.getUploadedFileUrl(request, "example");
+    assertFalse(uploadedFileInfo.isPresent());
+  }
+
+  /** Checks if a file uploaded successfully in Blobstore returns BlobInfo. */
+  @Test
+  public void checks_ifBlobUploaded_returnsFileInfo() throws IOException {
+    Date creationDate = new Date();
+    List<FileInfo> files = new ArrayList<FileInfo>();
+    FileInfo uploadFile = new FileInfo("blob", creationDate, "receipt", 0L, "hash", "edith");
+    files.add(uploadFile);
+    Map<String, List<FileInfo>> fileInfos = new HashMap<>();
+    fileInfos.put("fileName", files);
+
+    when(blobstoreService.getFileInfos(request)).thenReturn(fileInfos);
+    Optional<List<FileInfo>> uploadedFileInfo =
+        receiptFileHandlerService.getUploadedFileUrl(request, "fileName");
+    assertTrue(uploadedFileInfo.isPresent());
+  }
+
+  /** Checks if a file did not upload in Blobstore throws Exception. */
+  @Test(expected = IllegalStateException.class)
+  public void checks_ifFileDidNotUpload_throwsException() throws IOException {
+    List<FileInfo> files = Collections.emptyList();
+    receiptFileHandlerService.getBlobKey(files);
+  }
+
+  /** Checks if a file uploaded successfully in Blobstore returns url. */
+  @Test
+  public void checks_ifFileUploaded_returnsBlobKey() throws IOException {
+    Date creationDate = new Date();
+    FileInfo uploadFile = new FileInfo("blob", creationDate, "receipt", 0L, "hash", "edith");
+    List<FileInfo> files = new ArrayList<FileInfo>();
+    files.add(uploadFile);
+    BlobKey receiptKey = new BlobKey("key");
+    when(blobstoreService.createGsBlobKey(uploadFile.getGsObjectName())).thenReturn(receiptKey);
+    BlobKey returnedKey = receiptFileHandlerService.getBlobKey(files);
+    assertTrue(returnedKey.equals(receiptKey));
```

Comment:
Generally it's recommended to split our tests by newlines into three diff components. The components are arrange, act, and assert. This means all your code for preparing the method your testing is in one chunk, followed by the method under test, followed by the assert.

In this case I think it would be something like:
```
    Date creationDate = new Date();
    FileInfo uploadFile = new FileInfo("blob", creationDate, "receipt", 0L, "hash", "edith");
    List<FileInfo> files = new ArrayList<FileInfo>();
    files.add(uploadFile);
    BlobKey receiptKey = new BlobKey("key");
    when(blobstoreService.createGsBlobKey(uploadFile.getGsObjectName())).thenReturn(receiptKey);

    BlobKey returnedKey = receiptFileHandlerService.getBlobKey(files);

    assertTrue(returnedKey.equals(receiptKey));

Reply:
Thanks for the explanation and code snippet.
Changes made.